src/main/resources/project/server/deployApplication.pl

```diff
@@ -0,0 +1,446 @@
+                    && $new_json_server_groups_discovery->{result}->{$_}->{deployment}->{$expected_deployment_name}
+                    && $new_json_server_groups_discovery->{result}->{$_}->{deployment}->{$expected_deployment_name}->{enabled}
+            } @new_all_server_groups;
+
+            if (@new_enabled_server_groups) {
+                $jboss->log_info("Summary: deployment '$expected_deployment_name' is assigned and enabled on the server group(s): @new_enabled_server_groups.");
+                $summary .= "\nEnabled on: " . join(",", @new_enabled_server_groups) . " server groups.";
+            }
+
+            my @new_disabled_server_groups = grep {
+                $new_json_server_groups_discovery->{result}->{$_}->{deployment}
+                    && $new_json_server_groups_discovery->{result}->{$_}->{deployment}->{$expected_deployment_name}
+                    && !$new_json_server_groups_discovery->{result}->{$_}->{deployment}->{$expected_deployment_name}->{enabled}
+            } @new_all_server_groups;
+            if (@new_disabled_server_groups) {
+                $jboss->log_info("Summary: deployment '$expected_deployment_name' is assigned and disabled on the server group(s): @new_disabled_server_groups.");
+                $summary .= "\nDisabled on: " . join(",", @new_disabled_server_groups) . " server groups.";
+            }
+        };
+        if ($@) {
+            $summary .= " Error occured when composing summary: $@";
+            $jboss->log_warning("Error occured when composing summary: $@");
+            $jboss->warning();
+        }
+        $jboss->set_property(summary => $summary);
+        $jboss->log_info("=======Finished: composing summary=======");
+        return;
+    }
+    else {
+        ########
+        # logic for standalone jboss
+        ########
+        $jboss->log_info("=======Started: deploying to standalone=======");
+        my $cli_command_deploy_to_standalone = qq/deploy --force /;
+        $cli_command_deploy_to_standalone .= qq/ --runtime-name=$param_runtime_name / if $param_runtime_name;
+        $cli_command_deploy_to_standalone .= qq/ --name=$param_deployment_name / if $param_deployment_name;
+        $cli_command_deploy_to_standalone .= $source_is_url ? qq/ $param_application_content_source_path / : qq/ "$param_application_content_source_path" /;
+        $cli_command_deploy_to_standalone .= qq/ $param_additional_options / if $param_additional_options;
+
+        run_command_with_exiting_on_error(
+            command => $cli_command_deploy_to_standalone,
+            jboss   => $jboss
+        );
+        $jboss->log_info("=======Finished: deploying to standalone=======");
+
+        my $summary = "Application '$expected_deployment_name' has been successfully deployed from '$expected_source_for_summary'";
+        $jboss->set_property(summary => $summary);
+        return;
+    }
+}
+
+sub run_command_and_get_json_with_exiting_on_error {
```

Comment:
Maybe we need to move this code to JBoss.pm?

Reply:
It is possible way to go, only one thing that I do not want to make JBoss.pm bigger for now before understanding of it main purpose.